### DISCUSSION TITLE
How can I get the start day to be considered when calculating the number of days in the whole month?

### DISCUSSION THREAD

**Discussion by BrunoLongo1** [js-fns/date-fns#3070] [Q&A]
Hello everybody ! Suppose I have the start date of 2022 - 01 - 01 and the end date of 2022 - 01 - 31. If I use the differenceInMonths function I get 0 as a result, as this function does not include the start day (01-01- 2022). How can I get a certain function that considers the start day, having the result for this interval as 1 ? that is, the function considers the start and end day and in this case it sees the full month, returning the result 1.

Thanks !

**Comment by fturmel** [comment]
Would simply adding a day to your end date before passing it to the diff function work?

```js
import { differenceInMonths, addDays } from "date-fns";

const earlier = new Date(2022, 0, 1);
const later = new Date(2022, 0, 31);

const res = differenceInMonths(addDays(later, 1), earlier);

console.log(res); // 1
```

**Reply by BrunoLongo1** [reply]
Thanks for the reply, but I need a function that returns me the number of complete months and the sum of the balance of days disregarding the complete months. Example:

start date 2022-05-16
end date 2022-07-29
result:

full months: 1 (month 06)

balance of days: 16 days in month 05 (2022-05-16 to 2022-05-31) and 29 days in month 07 2022-07-01 to 2022-07-29), total 45 days

Final result:

full months: 1

days balance: 45 days
